### PR TITLE
ORV2 638: Bulk Delete Changes

### PR DESCRIPTION
**Pull request by gchauhan-aot**
# Description

This PR includes changes for Bulk Delete Power Units and Trailers. And unit as well as e2e test cases.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A: Unit tested using postman.
- [ ] Test B: unit teste and E2E test cases

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend - Vehicles](https://onroutebc-296-backend-vehicles.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://onroutebc-296-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/onroutebc/actions/workflows/merge-main.yml)

### CODE REVIEW

File: backend/vehicles/src/modules/vehicles/power-units/entities/power-unit.entity.ts [hunk 1]
```diff
@@ -92,4 +92,12 @@ export class PowerUnit extends Base {
   })
   @Column({ type: 'integer', name: 'STEER_AXLE_TIRE_SIZE', nullable: true })
   steerAxleTireSize: number;
+
+  @AutoMap()
+  @ApiProperty({
+    example: '1',
+    description: 'The Company Id to which the Power Unit belongs',
+  })
+  @Column({ type: 'integer', name: 'COMPANY_ID', nullable: true })
```
Comment (praju-aot) [inline backend/vehicles/src/modules/vehicles/power-units/entities/power-unit.entity.ts:101]: Company Id to be mandatory

File: backend/vehicles/src/modules/vehicles/trailers/trailers.controller.ts [hunk 2]
```diff
@@ -122,4 +126,25 @@ export class TrailersController {
     }
     return { deleted: true };
   }
+
+  @Post('delete-requests')
+  @ApiBody({
```
Comment (praju-aot) [inline backend/vehicles/src/modules/vehicles/trailers/trailers.controller.ts:131]: Please Add Response similar to powerUnits

**Review by praju-aot — CHANGES_REQUESTED** [review]

File: backend/vehicles/src/modules/vehicles/power-units/entities/power-unit.entity.ts (hunk 1 above)
Comment (gchauhan-aot) [reply backend/vehicles/src/modules/vehicles/power-units/entities/power-unit.entity.ts:101]: Not doing as per discussion.

File: backend/vehicles/src/modules/vehicles/trailers/trailers.controller.ts (hunk 2 above)
Comment (gchauhan-aot) [reply backend/vehicles/src/modules/vehicles/trailers/trailers.controller.ts:131]: Done.

**Review by gchauhan-aot — COMMENTED** [review]
Accommodated Review Comments

File: backend/vehicles/src/modules/vehicles/power-units/entities/power-unit.entity.ts (hunk 1 above)
Comment (praju-aot) [reply backend/vehicles/src/modules/vehicles/power-units/entities/power-unit.entity.ts:101]: Will take up in the next PR as making it mandatory will break create company

**Review by praju-aot — APPROVED** [review]

**Pull request merged**